### PR TITLE
[incubator-kie-issues-1550] Add Transaction error handling when transaction is enabled

### PR DESCRIPTION
**Pull request by elguardian**
Closes: https://github.com/apache/incubator-kie-issues/issues/1550

depends on https://github.com/apache/incubator-kie-kogito-runtimes/pull/3749

### CODE REVIEW

**Review by martinweiler — APPROVED** [review]

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/2030 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-2030/1/display/redirect

**Test results:**
- PASSED: 393
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-2030/1/testReport/org.kie.kogito.examples/GreetRestIT/testBiDiStreamGetMultipleLanguagesError/">org.kie.kogito.examples.GreetRestIT.testBiDiStreamGetMultipleLanguagesError</a></summary>
1 expectation failed.<br/>JSON path message doesn't match.<br/>Expected: (a string containing "io.grpc.StatusRuntimeException" and a string containing "OUT_OF_RANGE")<br/>  Actual: [jsongreetbidistreamerror:6dde14b7-7c52-4f7b-9e1e-ea107090ab39 - SayHelloMultipleLanguagesError:[uuid=6]] -- Received an error through gRPC server stream with status: Status{code=OUT_OF_RANGE, description=null, cause=null}<br/>
</details>

**Review by pefernan — APPROVED** [review]

File: kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties [hunk 1]
```diff
@@ -42,3 +42,5 @@ mp.messaging.outgoing.kogito-variables-events.value.serializer=org.apache.kafka.
 mp.messaging.outgoing.kogito-processdefinitions-events.connector=smallrye-kafka
 mp.messaging.outgoing.kogito-processdefinitions-events.topic=kogito-processdefinitions-events
 mp.messaging.outgoing.kogito-processdefinitions-events.value.serializer=org.apache.kafka.common.serialization.StringSerializer
+
+kogito.transactionEnabled=false
```
Comment (yesamer) [inline kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: @elguardian IINW, the correct property is `kogito.processes.transactionEnabled` //cc @gitgabrio

Comment (martinweiler) [reply kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: @yesamer both formats of the property are correct, it's either a global or a per-subsystem setting:

```
kogito.transactionEnabled=false           # tx enabled globally
#kogito.processes.transactionEnabled=true # tx enabled for the processes subsystem
#kogito.usertasks.transactionEnabled=true # tx enabled for the usertasks subsystem
```

Comment (gitgabrio) [reply kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: Thanks @martinweiler 
Does this follow the same logic of rest generation/code-generation, regarding precedence of global variable vs per-subsystem ?

Comment (yesamer) [reply kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: @martinweiler Thank you :)

Comment (martinweiler) [reply kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: @gitgabrio I think the global setting has precedence over the per-subsystem ones

Comment (gitgabrio) [reply kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: @martinweiler 🤔 

https://github.com/apache/incubator-kie-kogito-runtimes/blob/b8b74df59c2c54c34fb049ecb7bc341853842991/kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/util/CodegenUtil.java#L88

https://github.com/apache/incubator-kie-kogito-runtimes/blob/b8b74df59c2c54c34fb049ecb7bc341853842991/kogito-codegen-modules/kogito-codegen-api/src/main/java/org/kie/kogito/codegen/api/context/KogitoBuildContext.java#L98

Maybe you are right, but I have the impression that global take precedence normally, but for transaction the logic maybe is reverted: worth to double check with some real experiment ? <- @yesamer FYI

Comment (martinweiler) [reply kogito-quarkus-examples/process-quarkus-example/src/main/resources/application.properties:46]: Thanks for digging into this @gitgabrio  - I agree, it is worth to double check it

**Pull request merged**